### PR TITLE
Add check for checking duplicated links in README file

### LINKED ISSUE #1592: Add extra validation to check if more than two links are the same
There should be a check in the validation to check if more than two links are the same. 
If they exist in the README the "merge" button should be blocked to merge the PR.

### PR DESCRIPTION
**Pull request by marekdano**
Add extra check into validate_links.py file to check if there are some duplicated links in the README file

I create an ignored list for some duplicated links which are not related to any API link. There is this link `https://github.com/davemachado/public-api` in the README listed twice. Should we ignore it too?

Closes #1592

### CODE REVIEW

**Comment by matheusfelipeog** [comment]
Very good. I will review 😆 

> There is this link https://github.com/davemachado/public-api in the README listed twice. Should we ignore it too?

Yes, sounds reasonable to me.

File: build/validate_links.py [hunk 1]
```diff
@@ -16,6 +21,29 @@ def parse_links(filename):
...
     return links
 
+def dup_links(links):
+    """Check for duplicated links"""
+    print(f'Checking for duplicated links...')
+    hasError = False
+    seen = {}
+    dupes = []
+
+    for x in links:
```
Comment (matheusfelipeog) [inline build/validate_links.py:31]: Isn't it better to make it explicit?
```suggestion
    for link in links:
```

File: build/validate_links.py [hunk 2]
```diff
@@ -16,6 +21,29 @@ def parse_links(filename):
...
+        if x in ignored_links:
+            continue
+
+        if x not in seen:
+            seen[x] = 1
+        else:
+            if seen[x] == 1:
+                dupes.append(x)
```
Comment (matheusfelipeog) [inline build/validate_links.py:39]: Many URLs contain the `/` character at the end. This section does not consider them.

This is returning at the moment:
``` bash
Found duplicated links: ['https://github.com/davemachado/public-api']
```

If you consider the character `/` at the end:
``` bash
Found duplicated links: ['https://mailboxlayer.com', 'https://github.com/davemachado/public-api', 'https://languagelayer.com']
```

The `str.rstrip()` function solves this.

```suggestion
        x = x.rstrip('/')
        if x in ignored_links:
            continue

        if x not in seen:
            seen[x] = 1
        else:
            if seen[x] == 1:
                dupes.append(x)
```

**Review by matheusfelipeog — CHANGES_REQUESTED** [review]

File: build/validate_links.py (hunk 2 above)
Comment (marekdano) [reply build/validate_links.py:39]: @matheusfelipeog Thanks for reviewing my PR and spotting an extra check with `/` character.

**Review by matheusfelipeog — APPROVED** [review]

**Comment by matheusfelipeog** [comment]
Great 😄 

Now wait @yannbertrand review to merge.

**Comment by yannbertrand** [comment]
I noticed this PR failed because there was a duplicate link detected 😅

It seems like there is already some validation done on https://github.com/public-apis/public-apis/blob/5dc97c529e0bdd23c03d5a1840445c9c2c356e2c/build/validate_format.py#L74

Is it possible to have only one verification rather than two? I'm not sure the old way did work as I don't remember seeing the error. It would be even better to have the list of all dupes you added in the logs. So I suggest to remove the old way.

**Comment by yannbertrand** [comment]
> > There is this link https://github.com/davemachado/public-api in the README listed twice. Should we ignore it too?
> 
> Yes, sounds reasonable to me.

:+1:

**Comment by marekdano** [comment]
> I noticed this PR failed because there was a duplicate link detected 😅
> 
> It seems like there is already some validation done on
> 
> https://github.com/public-apis/public-apis/blob/5dc97c529e0bdd23c03d5a1840445c9c2c356e2c/build/validate_format.py#L74
> 
> Is it possible to have only one verification rather than two? I'm not sure the old way did work as I don't remember seeing the error. It would be even better to have the list of all dupes you added in the logs. So I suggest to remove the old way.

Done @yannbertrand

**Review by yannbertrand — APPROVED** [review]
LGTM! Thank you

**Pull request merged**